src/components/filesystem/index.js

```diff
@@ -353,7 +352,7 @@ module.exports = {
             });
         },
 
-	updateSocial: function() {
+	updateSocial: function(callbackFunc) {
```

Comment:
Do you think this would be cleaner by making this function return a promise, then you can .then in external change?